### ISSUE TITLE
CoreFX System.Security

### ISSUE THREAD

**Issue by marek-safar** [mono/mono#7589]
One of few assemblies where we didn't even migrated to referencesource. We are missing some functionality and our code is not very linker friendly. There are also reported security vulnerabilities we didn't act on.

**Comment by marek-safar** [comment]
This one will require multiple steps. It's better to start with small types first before doing any large ones.

**Comment by filipnavara** [comment]
I think System.Security.Cryptography.Pkcs should be checked off, X509Certificates is now tracked in issue #9216.

**Comment by marek-safar** [comment]
@filipnavara System.Security assembly has additional X509Certificates types which I track here (similarly for mscorlib)

**Comment by filipnavara** [comment]
@marek-safar My bad, I missed that. However, CoreFX doesn't implement these types at the moment. They are Windows-specific and all functions referencing them were wiped-out with `throw new PlatformNotSupportedException()`.

I once tried to implement `X509Certificate2UI.DisplayCertificate` overloads, but eventually I abandoned the effort. On macOS the underlying APIs require the application to be properly initialized Cocoa application. Moreover the window handle is dependent on the System.Windows.Forms backend (at least for the common usage), which currently lacks the Cocoa implementation altogether. Essentially the API is non-portable unless you constrain it with specific technologies. The only sensible implementation is by spawning a separate process that will implement the UI. Not sure if that's worth the effort, but I can contribute code for that if desired...

**Comment by marek-safar** [comment]
I agree the *UI API is just poor Windows API, we will keep the minimalistic implementation we have for now.
